### PR TITLE
add setup.sh script and use Heroku snapshots to restore

### PR DESCRIPTION
**Pull request by hcli-work**
https://app.asana.com/0/1179330853755533/1198870084975605

Just a small script that builds and launches Canvas repo
Also added script to download latest Heroku snapshot. It's binary though, so it first pg_restores to the database and then creates a plain text snapshot. From there, it replaces the URLs in the snapshot for a dev environment along with some other fancy stuff that the original script did.

Test Plan: used it and then tried accessing via canvasweb:3000

### CODE REVIEW

**Comment by hcli-work** [comment]
this is not ready for review yet. found a little hitch that I need to fix.

**Comment by hcli-work** [comment]
ready now

**Review by sadleb — CHANGES_REQUESTED** [review]
Many of my comments from the equivalent platform PR apply, but the big one is adding the db dumps (both binary and SQL) to .dockerignore. The other repos use heroku build packs and the docker containers are only for dev, but this repo uses the docker images to build production so it’s important we don’t build things like the whole production database, into the images. It’ll blow the image sizes up by at least 6GB in addition to the obvious security implications. In fact, one of the reasons that I had db refreshes run off production snapshots that had already been sanitized to work in staging by the admin server cron job was because I didn’t want to let devs access real production DBs directly on their local machines. That’s a SysOps security no-no.

**Comment by hcli-work** [comment]
I think I'm missing something here about the security point. The approach I've been going with has been to download the snapshot and then restore it. The current approach is to do the same thing with the main differences being that the download came from S3 and the dump itself wasn't saved on disk. But the data itself was persisted into the db containers, which ends up just writing it on disk anyway.

File: docker-compose/scripts/db_gensqldump.sh [hunk 1]
```diff
@@ -0,0 +1,34 @@
+#!/bin/bash
+echo "Restoring binary dump to database..."
+docker-compose down
+docker volume rm canvas-lms_canvas-db
+docker-compose up -d canvasdb
+sleep 5
+
+docker-compose exec canvasdb pg_restore --clean --jobs 2 --no-acl --no-owner -U canvas -d canvas latest.dump
+docker-compose exec canvasdb pg_restore --clean --jobs 2 --no-acl --no-owner -U canvas -d canvas latest.dump
```
Comment (sadleb) [inline docker-compose/scripts/db_gensqldump.sh:9]: You cleaned this up in one of the related PRs. Is this still a mistake or do we really need to run the restore twice? Also, is `latest.dump` the default file name that running the Heroku download goes into? It's confusing why `db_gensqldump.sh` starts with a restore. Either comment why or rename the script to be more appropriate? From what I can gather, this is just migrating the latest dump to a dev DB?

File: docker-compose/scripts/db_gensqldump.sh [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+echo "Recreating database snapshot with a SQL dump..."
+docker-compose exec canvasdb pg_dump --clean -U canvas canvas > latest-tmp.sql
```
Comment (sadleb) [inline docker-compose/scripts/db_gensqldump.sh:11]: Future readers aren't going to understand what is going on here at first glance. This is necessary b/c the Heroku dump is a binary dump and we need to turn it into a .sql dump, correct? If so maybe just clarify the above echo to be something like "Migrating binary database snapshot to SQL snapshot"?

File: docker-compose/scripts/db_gensqldump.sh [hunk 3]
```diff
@@ -0,0 +1,34 @@
...
+
+echo "Replacing production URLs with dev URLs...piping to database"
```
Comment (sadleb) [inline docker-compose/scripts/db_gensqldump.sh:13]: You're not piping it to the database from what I can tell.

**Review by sadleb — CHANGES_REQUESTED** [review]

File: docker-compose/scripts/db_gensqldump.sh (hunk 1 above)
Comment (hcli-work) [reply docker-compose/scripts/db_gensqldump.sh:9]: good catch. you're right. i cleaned it up in the actual restore. this is here just to convert the binary dump into a SQL one. i'll clean up here as well and then add some comments as to why it's needed.

File: docker-compose/scripts/db_gensqldump.sh (hunk 2 above)
Comment (hcli-work) [reply docker-compose/scripts/db_gensqldump.sh:11]: +1

File: docker-compose/scripts/db_gensqldump.sh (hunk 3 above)
Comment (hcli-work) [reply docker-compose/scripts/db_gensqldump.sh:13]: +1

**Review by sadleb — APPROVED** [review]
lgtm

**Pull request merged**